tests/QE/test_system.py

```diff
@@ -0,0 +1,597 @@
+
+        # Execute all CRUD tasks concurrently using asyncio
+        self.mark_test_step(
+            "Executing concurrent CRUD operations - each database operates independently"
+        )
+        try:
+            # Run all database CRUD loops concurrently until time expires
+            await asyncio.gather(
+                *[crud_task for _, crud_task in crud_tasks], return_exceptions=True
+            )
+        except Exception as e:
+            print(f"Exception in concurrent execution: {e}")
+
+        self.mark_test_step("CRUD operations loop completed - verifying final state")
+        total_local_count = 0
+        for db_name, db in databases.items():
+            final_local_docs = await db.get_all_documents("_default._default")
+            local_count = len(final_local_docs["_default._default"])
+            total_local_count += local_count
+            print(f"Final document count in {db_name}: {local_count}")
+
+        final_sg_docs = await cblpytest.sync_gateways[0].get_all_documents("names")
+        sg_count = len(final_sg_docs.rows)
+
+        print(
+            f"Final document counts - Total CBL: {total_local_count}, SGW: {sg_count}"
+        )
+        print(f"Test duration: {datetime.now() - self.iteration_start_time}")
+        print(
+            f"SUCCESS: Concurrent test completed exactly {self.UP_TIME_DAYS} day(s) of continuous CRUD operations"
+        )
+
+        # Verify replication consistency (allowing for some tolerance due to concurrent operations and network issues)
+        tolerance = max(
+            100, total_local_count * 0.02
+        )  # 2% tolerance or minimum 100 docs (higher due to concurrency)
+        assert abs(total_local_count - sg_count) <= tolerance, (
+            f"Document count mismatch between CBL ({total_local_count}) and SGW ({sg_count}). Difference should be <= {tolerance}."
+        )
+
+        print(
+            "SUCCESS: Concurrent execution maintained document consistency between CBL apps and Sync Gateway"
+        )
+
+        # Cleanup all test servers dynamically
+        for i, test_server in enumerate(cblpytest.test_servers):
+            await test_server.cleanup()
+            print(f"Cleaned up test server {i}")
+
+    async def _wait_for_replication_idle_single(
+        self, replicator: Replicator, db_name: str
+    ) -> None:
+        """Helper method to wait for a single replicator to reach idle state"""
+        try:
+            status = await replicator.wait_for(
+                ReplicatorActivityLevel.IDLE,
+                timedelta(seconds=self.REPL_STATUS_CHECK_SLEEP_TIME),
+                timedelta(minutes=3),  # Shorter timeout for individual threads
+            )
+            if status.error:
+                print(
+                    f"[{db_name}] Warning: Replication error during wait: {status.error}"
+                )
+        except Exception as e:
+            print(f"[{db_name}] Warning: Error waiting for replication idle: {e}")
```

Comment:
What is the point of doing this?  This will make this error harder to find since it will only be present in the jenkins console, and when the test fails it will fail for a different reason.